core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -355,11 +355,85 @@ public static boolean regexpContains(String value, String regex) {
       Pattern regexp = Pattern.compile(regex);
       return regexp.matcher(value).find();
     } catch (PatternSyntaxException ex) {
-      throw RESOURCE.invalidInputForRegexpContains(ex.getMessage().replace("\r\n", " ")
-          .replace("\n", " ").replace("\r", " ")).ex();
+      throw RESOURCE.invalidRegexInputForRegexpFunctions(ex.getMessage().replace("\r\n", " ")
+              .replace("\n", " ").replace("\r", " "),
+          "REGEXP_CONTAINS").ex();
     }
   }
 
+  /** SQL {@code REGEXP_EXTRACT(value, regexp[, position[, occurrence]])} function.
+   *  Returns NULL if there is no match, or if position or occurrence are beyond range.
+   *  Returns an exception if regex, position or occurrence are invalid.*/
+  public static @Nullable String regexpExtract(String value, String regex, Integer... params) {
+    return processRegexpExtractOrSubstr("REGEXP_EXTRACT", value, regex, params);
+  }
+
+  /** SQL {@code REGEXP_SUBSTR(value, regexp[, position[, occurrence]])} function.
+   *  Returns NULL if there is no match, or if position or occurrence are beyond range.
+   *  Returns an exception if regex, position or occurrence are invalid.*/
+  public static @Nullable String regexpSubstr(String value, String regex, Integer... params) {
+    return processRegexpExtractOrSubstr("REGEXP_SUBSTR", value, regex, params);
+  }
+
+  private static @Nullable String processRegexpExtractOrSubstr(String methodName, String value,
```

Comment:
once you add one as the alias of the other you can just change this to `regexpExtract` to match other functions